Source/WarpX.cpp

```diff
@@ -2030,6 +2048,18 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
     AllocInitMultiFab(current_fp[lev][1], amrex::convert(ba, jy_nodal_flag), dm, ncomps, ngJ, tag("current_fp[y]"), 0.0_rt);
     AllocInitMultiFab(current_fp[lev][2], amrex::convert(ba, jz_nodal_flag), dm, ncomps, ngJ, tag("current_fp[z]"), 0.0_rt);
 
+    // Match external field MultiFabs to fine patch
+    if (add_external_B_field) {
+        AllocInitMultiFab(Bfield_fp_external[lev][0], amrex::convert(ba, Bx_nodal_flag), dm, ncomps, ngEB, tag("Bfield_fp_external[x]"));
+        AllocInitMultiFab(Bfield_fp_external[lev][1], amrex::convert(ba, By_nodal_flag), dm, ncomps, ngEB, tag("Bfield_fp_external[y]"));
+        AllocInitMultiFab(Bfield_fp_external[lev][2], amrex::convert(ba, Bz_nodal_flag), dm, ncomps, ngEB, tag("Bfield_fp_external[z]"));
+    }
+    if (add_external_E_field) {
+        AllocInitMultiFab(Efield_fp_external[lev][0], amrex::convert(ba, Ex_nodal_flag), dm, ncomps, ngEB, tag("Efield_fp_external[x]"));
+        AllocInitMultiFab(Efield_fp_external[lev][1], amrex::convert(ba, Ey_nodal_flag), dm, ncomps, ngEB, tag("Efield_fp_external[y]"));
+        AllocInitMultiFab(Efield_fp_external[lev][2], amrex::convert(ba, Ez_nodal_flag), dm, ncomps, ngEB, tag("Efield_fp_external[z]"));
+    }
+
```

Comment:
I wonder if this causes a segfault in case one tries to read external fields with more than one MR level.

I saw you assert that in `WarpX::InitLevelData` in Source/Initialization/WarpXInitData.cpp.

However, the order of function calls seems to me to be the following:
- `AmrCore::InitFromScratch`
  - `WarpX::MakeNewLevelFromScratch(lev, ...)`
    - `AllocLevelData(lev, ...)`
      - `AllocLevelMFs(lev, ...)`
      (which attempts to allocate the external field MultiFabs for a given level)
    - `InitLevelData(lev, ...)`
    (which asserts that external fields cannot be read with more than one level)

It seems to me that, with more than one MR level, one could encounter a segfault before the assert message is triggered. Is it possible that this is the case or am I missing something? May be worth double checking.

Reply:
You're almost certainly right. I'll check and adjust those guards

Reply:
Ok, looking more closely at this, I'm not seeing definite segfault potential, other than that these calls add the allocation footprint of one more MultiFab over the same boxes as the rest of the simulation already allocates. 

That said, I'll move the assertions here or to parameter reading, so we don't even risk that.